src/Merchello.Core/Extensions.ExtendedDataCollection.cs

```diff
@@ -787,8 +782,17 @@ public static IAddress GetAddress(this ExtendedDataCollection extendedData, stri
         {
             if (!extendedData.ContainsKey(dictionaryKey)) return null;
 
-            var attempt = SerializationHelper.DeserializeXml<Address>(extendedData.GetValue(dictionaryKey));
+            // Get the address data from the extendedData
+            var addressData = extendedData.GetValue(dictionaryKey);
 
+            // Firstly try JSON convert
+            // TODO - Should really make this is a bit more fool proof
+            if (addressData.Contains("{"))
```

Comment:
How about using something like this - https://stackoverflow.com/questions/14977848/how-to-make-sure-that-string-is-valid-json-using-json-net